pkg/controllers/kube-storage-version-trigger.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright © 2021 MicroShift Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util"
+
+	"k8s.io/client-go/tools/clientcmd"
+	klog "k8s.io/klog/v2"
+	migrationclient "sigs.k8s.io/kube-storage-version-migrator/pkg/clients/clientset"
+	"sigs.k8s.io/kube-storage-version-migrator/pkg/trigger"
+)
+
+const (
+	kubeStorageVersionTrigger = "storage-version-migration-trigger"
+)
+
+type KubeStorageVersionTrigger struct {
+	kubeconfig string
+	healthPath string
+	healthPort string
+}
+
+func NewKubeStorageVersionTrigger(cfg *config.Config) *KubeStorageVersionTrigger {
```

Comment:
What's the difference between the trigger and the migrator?

Reply:
There's [user guide](https://github.com/openshift/kubernetes-kube-storage-version-migrator/blob/master/USER_GUIDE.md#storage-version-migrator-in-a-nutshell) for it'd be nice to include that information in our repository (even as a comment in one of the files).

Looks like we'll need to include [checking if migration is complete and successful](https://github.com/openshift/kubernetes-kube-storage-version-migrator/blob/master/USER_GUIDE.md#check-if-migration-has-completed) in healthcheck script

Reply:
And we'll need to rethink `$DATA_DIR/version`  - current impl was based on assumption that migration would happen before "true" start, but with this we'll just start and healthcheck will assess if all is good.
Maybe it's a good thing and will simplify and increase flexibility of the upgradability. :thinking: